update deployment scenarios (#1437)
Azure/iotedge-lorawan-starterkit: docs/user-guide/deployment-scenarios.md

![Keep data on-premise](../images/scenarios/scenario-local-data.png)

## Scalability & Availability

### Single conentrator with single LNS

This is the most basic deployment you can chose. It does only deploy 1 concentrator

> per and one network server. Each sensor can at most reach a single concentrator.

Reviewer: ```suggestion
and one network server. Each sensor can at most reach a single concentrator.
```

#### Advantages

1. Scalability: this model scales very well. There is no additional overhead to deduplicate
messages, nor do we need to determine who is winning the race to potentially send messages

> back to the device. Also, no additional edge hub does keep a connection for the device - which helps with the scale requirement on the IoT Hub itself.

Reviewer: ```suggestion
back to the device. Also, no additional edge hub keeps a connection for the leaf device, which helps with the scale requirement on the IoT Hub itself.
```

1. Simple: the deployment is simple to start with and maintain.
1. Low cost: No redundant hardware is used, which helps keeping the operational cost down.

#### Disadvantages

> 1. Availability: this solution has 2 single point of failure: the concentrator and the

Reviewer: ```suggestion
1. Availability: this solution has 2 single points of failure: the concentrator and the
```

network server. Either of those can go down. The result in the worst case is lost
of messages.
1. Limited reach: Since there is only a single concentrator, you have to put all your sensor
within reach of that concentrator.
1. Reliability: message delivery might be impacted.

#### Recommended settings

| Target        | Setting               |Value|
|---------------|:----------------------|:--------|
| Device Twin   |KeepAliveTimeout       |undefined or 0 |
| Device Twin   |Deduplication          |None           |

> | Device Twin   |GatewayId              |The Id of the Newtork Server |

Author: already fixed

> ![Single Single](../images/scenarios/1_scale_and_availability.jpg)

Reviewer: ```suggestion
![Single concentrator - Single gateway](../images/scenarios/1_scale_and_availability.jpg)
```

### Multi conentrator with single LNS

Multiple concentrators are deployed that reach a single network server. Each sensor can reach one or multiple concentrators. Depending on the goal of the deployment (availability and/or reach).

#### Advantages

1. Scalability: this model scales very well. It's a bit more overhead than the single/single
model as it will have to deduplicate messages coming in from the additional
concentrators.
1. Relatively simple
1. Low cost
1. Partial redundancy: if the deployment is designed to ensure that each sensor can reach
at least 2 concentrators, you get higher availability than with single/single.

> 1. Reach: if you do deploy for reach, you can get a larger support a larger geographical

Reviewer: ```suggestion
1. Reach: if you deploy for reach, you can get a larger support a larger geographical
```

deployment. To combine reach with availability, the cost will raise.

#### Disadvantages

1. Availability: Depending on the deployment, this solution has at least 1 single point of failure: the network server. The result in the worst case is lost of messages.
1. Reliability: message delivery might be impacted.

#### Recommended settings

| Target        | Setting               |Value|
|---------------|:----------------------|:--------|
| Device Twin   |KeepAliveTimeout       |undefined or 0 |
| Device Twin   |Deduplication          |None           |

> | Device Twin   |GatewayId              |The Id of the Newtork Server |

Reviewer: ```suggestion
| Device Twin   |GatewayId              |The Id of the Network Server |
```

Author: already fixed

![Multi Single](../images/scenarios/2_scale_and_availability.jpg)

> ### Multi conentrator with Multi LNS

Reviewer: ```suggestion
### Multiple conentrator with multiple LNS
```

can only be connected to a single LNS). Each sensor can reach one or multiple concentrators. Depending on the goal of the deployment (availability and/or reach).

To get high availability concentrator deployment and assignment to LNS should be done
so that a single sensor is in range of at least 2 concentrators, both connecting to a
**different** LNS.

#### Advantages

1. Availability: If the deployment is done right, the single point of failures can
be eliminated.
1. Reliability: message delivery likelyhood is increased.
1. Reach: The deployment model can be expanded

#### Disadvantages

1. Cost: all components need to be deployed and managed multiple times.
1. Complexity
1. Scalability: The scalability is impacted as the LNS now needs to manage leader election
for message handling, as well as multiple connetion are opened for the same client on
multiple edge hubs, resulting in connection ping pong.

#### Recommended settings

| Target        | Setting               |Value|
|---------------|:----------------------|:--------|
| Device Twin   |KeepAliveTimeout       | > 0 - depending on the sensor, this can be tuned

> to be the time you expect a new message to come in. Ideally not > 300 |

Reviewer: Should we give explanations here about the reason why it should not be > 300?

Author: not in the table I believe. We can do an explanation of the issue somewhere.

Author: as discussed, let's create an ADR for the next iteration on the connection management changes and explain it there.

> | Device Twin   |Deduplication          |Drop           |

Reviewer: Should we give explanations why Mark/None could give issues?

Author: same here as above.

![Keep data on-premise](../images/scenarios/scenario-local-data.png)

## Scalability & Availability

> ### Single conentrator with single LNS

Reviewer 2: ```suggestion
### Single concentrator with single LNS
```

This is the most basic deployment you can chose. It does only deploy 1 concentrator
and one network server. Each sensor can at most reach a single concentrator.

#### Advantages

1. Scalability: this model scales very well. There is no additional overhead to deduplicate
messages, nor do we need to determine who is winning the race to potentially send messages
back to the device. Also, no additional edge hub keeps a connection for the leaf device, which helps with the scale requirement on the IoT Hub itself.
1. Simple: the deployment is simple to start with and maintain.
1. Low cost: No redundant hardware is used, which helps keeping the operational cost down.

#### Disadvantages

> 1. Availability: this solution has 2 points of failure: the concentrator and the

Reviewer 2: Consider rephrasing it to two additional points of failure, since there may also be other points of failure in our system (Function, IoT Hub, Redis, etc).

Author: added a general note on the other services. This is only focusing on LBS/LNS. Other services have dedicated documentation on HA and scalability, that should be taken into considation.

c687a555748d05d84a33a4555a06c878e2cc1df8
